### DISCUSSION TITLE
Example for a sumup in table footer

### DISCUSSION THREAD

**Discussion by capsloq** [glideapps/glide-data-grid#424] [Q&A]
Hello,

just discovered this project. Looks very promising, the speed is insane.
Since i didn't find it in the storybook:
Is it possible to showcase a table footer component which sums up a column of numbers?
(Ideally the sum changes whenever a number in that column is changed)

Thank you.

**Comment by jassmith** [comment]
I have to admit, I am confused why you would want this to be a function this component provides? Summing over your data is trivial, by necessity you must be the one that informs the DataGrid when data changes. It would be very straightforward to build this but I don't get why...

Even if I did build this it would just be an additional package which consumes the same callbacks as the data-editor itself and does simple math for you.

**Reply by capsloq** [reply]
Hello jassmith,

thank you for the quick response.
It may be trivial for you. However when I checked out the storybook examples I was looking for some example where calculations are included. Meaning the value of specific cells being based on others. I was hoping to see like a best practice looping over row values, column values and so on. In addition I asked for displaying those results in the table footer because I didn't see any example which had one.
Even though it may be  trivial to people familiar with the project / API, me personally could not figure it out on my own so it might help others as well.

**Reply by jassmith** [reply]
I will make a demo.

**Reply by jassmith** [reply]
Oh just to be clear, I wasn't trying to demean you! Just confused why a footer would be a desired feature. I can make a demo of a self built footer and we can decide if we want to turn it into a package later.

**Reply by capsloq** [reply]
All good. I was having a first look at different data grids and evaluated them quickly. So i had limited time to check out each project. While glide-data-grid has its own awesome way of doing things, it was lacking some showcasing for my specific use case. That's why i opened this discussion mostly to improve documentation / storybook examples. However you were right I had a second look today and it is pretty clear on how to achieve most of the stuff i was asking for. However i am looking forward to your example. Maybe it can even become some sort of kitchen sink for common calculations in the future. Regarding the tablefooter i am having a usecase similar to this ![Screenshot](https://user-images.githubusercontent.com/12716713/27181773-b0ae783e-51e1-11e7-9323-843fcdd3b12d.png)

**Reply by jassmith** [reply]
I see what you want, okay this is pretty straightforward to do. Definitely something we should add as a hook to the glide-data-grid-source package.
